### PR TITLE
Create kyaml command to expose running kyaml filters on the cli

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: cmd/kyaml/cmd/merge.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Merge reads Kubernetes Resource yaml configuration files from stdin or sources packages and write
+the result to stdout or a destination package.
+
+Resources are merged using the Resource [apiVersion, kind, name, namespace] as the key.  If any of
+these are missing, merge will default the missing values to empty.
+
+Resources specified later are high-precedence (the source) and Resources specified
+earlier are lower-precedence (the destination).
+
+Merge uses the following rules for merging a source Resource into a destination Resource:
```
Comment (bzub) [inline cmd/kyaml/cmd/merge.go]: Does this line need to be moved or is this a TODO?

File: cmd/kyaml/cmd/tree.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	c.Flags().BoolVar(&r.cmd, "command", false, "print command field")
+	c.Flags().BoolVar(&r.env, "env", false, "print env field")
+	c.Flags().BoolVar(&r.all, "all", false, "print all field infos")
+	c.Flags().StringSliceVar(&r.fields, "field", []string{}, "print field")
+	c.Flags().BoolVar(&r.includeReconcilers, "include-reconcilers", false,
+		"if true, include reconciler Resources in the output.")
+	c.Flags().BoolVar(&r.excludeNonReconcilers, "exclude-non-reconcilers", false,
+		"if true, exclude non-reconciler Resources in the output.")
+	c.Flags().StringVar(&r.structure, "structure", "package",
+		"structure to use for the tree.  may be 'package' or 'owners'.")
```
Comment (bzub) [inline cmd/kyaml/cmd/tree.go]: Missing `graph`?

File: cmd/kyaml/cmd/merge.go (hunk 1 above)
Comment (pwittrock) [reply cmd/kyaml/cmd/merge.go]: Should be removed and followed up on:

https://github.com/kubernetes-sigs/kustomize/issues/1756

File: cmd/kyaml/cmd/tree.go (hunk 2 above)
Comment (pwittrock) [reply cmd/kyaml/cmd/tree.go]: Done

File: cmd/kyaml/cmd/merge.go (hunk 1 above)
Comment (pwittrock) [reply cmd/kyaml/cmd/merge.go]: Added

**Pull request merged**